client-app/src/desktop/common/grid/SampleTreeGridModel.js

```diff
@@ -185,7 +185,6 @@ export class SampleTreeGridModel {
         const {store} = this,
             isChecked = !rec.data.isChecked,
             updates = [
```

Comment:
seems to me, we might need the first bit for a leaf node, no?  Did you try checking a leaf node ???  :-)

Reply:
I think this is ok because our `allDescendants` will always return the record, even if it has no children (if I am reading the code correctly).

Reply:
great -- need to start reading the entire file -- sorry for muddying the waters here.